src/plugins/data_source_management/public/types.ts

```diff
@@ -62,3 +75,21 @@ export interface ToastMessageItem {
 export type DataSourceManagementContextValue = OpenSearchDashboardsReactContextValue<
   DataSourceManagementContext
 >;
+
+export enum CredentialSourceType {
+  CreateCredential = 'createCredential',
+  ExistingCredential = 'existingCredential',
+  NoAuth = 'noAuth',
+}
+
+export interface CreateNewCredentialType {
+  title: string;
+  description: string;
+  credentialMaterials: {
+    credentialMaterialsType: string;
```

Comment:
instead of string, let's use CredentialMaterialsType.UsernamePasswordType for the keywords

Reply:
will make these changes with new design change